dlc-manager/src/sub_channel_manager.rs

```diff
@@ -2741,8 +2824,64 @@ where
+                                })?;
+                            let dlc_channel = get_channel_in_state!(
+                                self.dlc_channel_manager,
+                                &dlc_channel_id,
+                                Signed,
+                                None::<PublicKey>
+                            )?;
+                            let contract = get_contract_in_state!(
+                                self.dlc_channel_manager,
+                                &dlc_channel
+                                    .get_contract_id()
+                                    .expect("Signed contract should have a contract id"),
+                                Signed,
+                                None::<PublicKey>
+                            )?;
+                            let offered_channel = OfferedChannel {
+                                offered_contract_id: contract.accepted_contract.offered_contract.id,
+                                temporary_channel_id: dlc_channel.temporary_channel_id,
+                                party_points: dlc_channel.own_points,
+                                per_update_point: dlc_channel.own_per_update_point,
+                                offer_per_update_seed: channel.per_split_seed,
+                                is_offer_party: false,
+                                counter_party: dlc_channel.counter_party,
+                                // TODO(tibo): use value from original offer
+                                cet_nsequence: CET_NSEQUENCE,
+                            };
+                            self.ln_channel_manager.set_funding_outpoint(
+                                channel_lock,
+                                &a.ln_rollback.funding_outpoint,
+                                a.ln_rollback.channel_value_satoshis,
+                                a.ln_rollback.value_to_self_msat,
+                            );
+                            self.dlc_channel_manager.get_store().upsert_channel(
+                                Channel::Offered(offered_channel),
+                                Some(Contract::Offered(
+                                    contract.accepted_contract.offered_contract,
+                                )),
+                            )?;
+                            updated_state = Some(SubChannelState::Offered(OfferedSubChannel {
+                                per_split_point: a.own_per_split_point,
+                            }));
+                            Ok(())
+                        })
+                        .map_err(|e| Error::InvalidState(format!("{:?}", e)))?;
+                }
+                SubChannelState::Finalized(signed) => {
                     if let Some(counter_state) = peer_state {
-                        if counter_state == ReestablishFlag::Accepted as u8 {
+                        if counter_state == ReestablishFlag::Signed as u8 {
```

Comment:
Here we are saying that we can assume the channel as signed if the other party says they have it as signed (presumably because they actually received our finalize message). 

I guess this implies they've also sent us (via the channel reestablish) the revocation keys that they would've normally sent via the new revoke message. Otherwise I think it wouldn't be a good idea to proceed, since the other party could always publish the old commitment transaction. Does that make sense?

Reply:
Yes that's it. Basically the LN protocol will make sure that we have the needed revocation secret after reestablishment.